### PR TITLE
12) added user model and connected with it staff, implement authentic…

### PR DESCRIPTION
**Pull request by AntonPanchishko**
…ation service

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.hibernate.query.Query;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        byte [] salt = HashUtil.getSalt();
+        user.setPassword(HashUtil.getHashPassword(user
+                .getPassword(), salt));
+        user.setSalt(salt);
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: This should be done in [UserService](https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_05.html). It's not a part of DAO

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> userQuery = session
+                    .createQuery("from User u where u.email =:email",
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: why do you need alias here?

File: src/main/java/com/dev/cinema/model/User.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    @Override
+    public String toString() {
+        return "User{" + "id=" + id + ", email='" + email + '\''
+                + ", password='" + password + '\'' + '}';
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/model/User.java]: You heard the man, there is a standard way to override basic methods.

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            throw new AuthenticationException("Can't find user with such email " + email);
+        }
+        User userByEmail = userService.findByEmail(email).get();
+        String passwordHashFromDB = userByEmail.getPassword();
+        String hashOfInputPassword = HashUtil.getHashPassword(password, userByEmail.getSalt());
+        if (passwordHashFromDB.equals(hashOfInputPassword)) {
+            return userByEmail;
+        }
+        throw new AuthenticationException("Cant find user with such "
+                + email + " and password " + password);
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]: This is a matter of security, do not specify what did user input wrong here. In fact, if you show this message in the normal system - your password will be open for prying eyes.

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -25,8 +25,7 @@ public User login(String email, String password) throws AuthenticationException
         if (passwordHashFromDB.equals(hashOfInputPassword)) {
             return userByEmail;
         }
-        throw new AuthenticationException("Cant find user with such "
-                + email + " and password " + password);
+        throw new AuthenticationException("Cant find user with such email and password");
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]: lets throw this ones. HINT combine checks

**Review by Max-Pochepets — APPROVED** [review]
👍

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        if (optionalUser.isEmpty() || !HashUtil.getHashPassword(password,
+                optionalUser.get().getSalt())
+                .equals(optionalUser.get().getPassword())) {
+            throw new AuthenticationException("Can't find user with such email " + email);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java:22]:
```suggestion
            throw new AuthenticationException("Can't find a user with such email or password");
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**